common/models/model-method.js

```diff
@@ -10,4 +11,48 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+
+  ModelMethod._shouldEncodeStaticFlagInName = function() {
+    var version = ModelMethod.app.models.Workspace.loopBackVersion;
+    return version != null ? !semver.gtr('3.0.0', version) : false;
+  };
+
+  ModelMethod.getJsonKey = function(name, data) {
+    if (!this._shouldEncodeStaticFlagInName()) {
+      return data.name;
+    }
+    var isStatic = data.isStatic;
+    if (isStatic !== undefined) {
+      var matchName = name.match(/^prototype\.(.*)$/);
+      if (!isStatic && (matchName === null || !matchName)) {
+        data.name = 'prototype.' + name;
+      }
+    }
+
+    return data.name;
+  };
+
+  ModelMethod.getConfigFromData = function(data) {
+    var config = ModelMethod.base.getConfigFromData.call(this, data);
+
+    if (this._shouldEncodeStaticFlagInName()) {
+      delete config.isStatic;
+    }
+
+    return config;
+  };
+
+  ModelMethod.getDataFromConfig = function(config, name) {
+    var data = ModelMethod.base.getDataFromConfig.call(this, config);
+    data.name = name;
+
+    if (this._shouldEncodeStaticFlagInName()) {
```

Comment:
I see. I was expecting that `keySetter` would be a function that converts `prototype.updateAttributes` key into `name:updateAttributes` & `isStatic:false`, while you are taking care of that conversion directly here, in `getDataFromConfig`. Neat!